index.d.ts

```diff
@@ -0,0 +1,23 @@
+/// <reference types="node"/>
+import {Duplex, DuplexOptions, Readable, Writable} from 'stream';
+
+export interface DuplexifyOptions extends DuplexOptions {
+  autoDestroy?: boolean;
+  end?: boolean;
+}
+
+export class Duplexify extends Duplex {
```

Comment:
This should be an `interface` or you should write this all as a `class`. The tests worked fine for the sub-classing case so I'd love to know more if that's the reason you used `class` here.

Reply:
Nope! I have no real reasoning for why I chose `class`, so I'm happy to change it to an `interface`.